tests/test_connection_manager.py

```diff
@@ -96,6 +100,21 @@ def test_boto_session_with_iam_role_and_no_cache(
         )
 
     @patch("sceptre.connection_manager.boto3.session.Session")
+    @patch("sceptre.connection_manager.boto3.client")
+    def test_boto_session_with_iam_profile_and_no_cache(
+            self, mock_client, mock_Session
+    ):
+        mock_Session.return_value = sentinel.session
+        self.connection_manager.iam_profile = sentinel.iam_profile
+        boto_session = self.connection_manager.boto_session
+
+        assert boto_session == sentinel.session
+        mock_Session.assert_called_once_with(
+            profile_name=self.connection_manager.iam_profile,
+            region_name=self.region
+        )
+
+    @patch("sceptre.connection_manager.boto3.session.Session")
```

Comment:
Can we test that a profile is not used when `connection_manager.iam_profile=None`?